main.js

```diff
@@ -46,6 +48,14 @@ const animate = () => {
   drawPaddles();
   drawBall();
   updatePaddles();
+  ball.x += ballSpeedHorizontal;
+  ball.y += ballSpeedVertical;
+  if (ball.x < 10 || ball.x > 949) {
+    ballSpeedHorizontal *= -1;
+  }
+  if (ball.y < 10 || ball.y > 539) {
+    ballSpeedVertical *= -1;
+  }
```

Comment:
Unlike updating the position of the ball, which can be cleanly contained in the ball class, this is an interaction between two object, so has to be brought up a level in the abstraction tree.

Since this relates to the ball position it should go in the `updateBall` function already defined and called